### PR TITLE
feat: Implement calendar API for export member events to ics file

### PR DESCRIPTION
**Pull request by jhihruei**
#### Introduce 
- Create an API to return member's events in ics format
- Ensure the API response could be added to Google Calendar
- Calendar API endpoint: `GET /api/v2/calendars/:memberId` 
- Ensure the calendar events will not duplicate after re-syncing.

### CODE REVIEW

**Comment by clothe09986** [comment]
[【外包reivew】Calendar & Meta Lead Ad](https://trello.com/c/SVlmsVxO/4938-%E3%80%90%E5%A4%96%E5%8C%85reivew%E3%80%91calendar-meta-lead-ad)

File: src/member/member.service.ts [hunk 1]
```diff
@@ -342,4 +343,11 @@ export class MemberService {
   async deleteMemberByEmail(appId: string, email: string): Promise<DeleteResult> {
     return this.memberInfra.deleteMemberByEmail(appId, email, this.entityManager);
   }
+
+  async getMemberTasks(memberId: string): Promise<Array<MemberTask>> {
+    const tasks = this.entityManager.getRepository(MemberTask).find({
+      where: { member: { id: memberId } },
+    });
```
Comment (clothe09986) [inline src/member/member.service.ts]: Business logic is written in the service layer, and data queries are written in the infra layer. So it should be written in `member.infra.ts`.

File: src/order/order.service.ts [hunk 2]
```diff
@@ -576,4 +576,16 @@ export class OrderService {
...
+
+  public async getOrderProductsByMemberId(memberId: string, productType?: string): Promise<OrderProduct[]> {
+    const orderProducts = await this.entityManager.getRepository(OrderProduct).find({
+      where: {
+        order: {
+          memberId: memberId,
+        },
+        ...(productType && { product: { type: productType } }),
+      },
+    });
```
Comment (clothe09986) [inline src/order/order.service.ts]: Business logic is written in the service layer, and data queries are written in the infra layer. So it should be written in `order.infra.ts`.

**Review by clothe09986 — CHANGES_REQUESTED** [review]

File: src/member/member.service.ts (hunk 1 above)
Comment (jhihruei) [reply src/member/member.service.ts]: updated, thanks for reminder.

File: src/order/order.service.ts (hunk 2 above)
Comment (jhihruei) [reply src/order/order.service.ts]: updated, thanks for reminder.

**Comment by jhihruei** [comment]
rebase to resolve conflict.

File: src/calendar/calendar.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      return JSON.parse(result);
+    });
+    if (cachedEvents) {
+      return cachedEvents;
+    }
+
+    const tasks = await this.memberService.getMemberTasks(memberId);
+    const taskEvents: EventAttributes[] = tasks.map((task) => {
+      return {
+        uid: task.id,
```
Comment (jhihruei) [inline src/calendar/calendar.service.ts:29]: Add `uid` to ensure the calendar events will not duplicate when re-syncing the calendar events.

File: src/calendar/calendar.service.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        title: task.title,
+        description: task.description || '',
+        duration: { minutes: 0 },
+      };
+    });
+
+    const orderProducts = await this.orderService.getOrderProductsByMemberId(memberId, 'AppointmentPlan');
+    const orderProductEvents: EventAttributes[] = orderProducts.map((orderProduct) => {
+      return {
+        uid: orderProduct.id,
```
Comment (jhihruei) [inline src/calendar/calendar.service.ts:40]: Add `uid` to ensure the calendar events will not duplicate when re-syncing the calendar events.

**Comment by jhihruei** [comment]
@clothe09986 Add two commits after rebase. 
- Add `uid` to calendar events to prevent the events from duplicating after re-syncing.
- Refactor to use the ics built-in function to transfer `Date` to ics `DateArray`.

**Review by clothe09986 — APPROVED** [review]

**Pull request merged**